prov/efa/src/efa_cq.c

```diff
@@ -540,6 +541,8 @@ static ssize_t efa_cq_sreadfrom(struct fid_cq *cq_fid, void *buf, size_t count,
 
 	for (num_completions = 0; num_completions < threshold; ) {
 		if (efa_cq_trywait(cq) == FI_SUCCESS) {
+			if (ofi_adjust_timeout(endtime, &timeout))
```

Comment:
Good catch, the code was referencing verbs provider https://github.com/ofiwg/libfabric/blob/main/prov/verbs/src/verbs_cq.c#L194 which likely has the same bug